examples/point_transformer_classification.py

```diff
@@ -0,0 +1,213 @@
+        return x
+
+
+class TransitionDown(torch.nn.Module):
+    '''
+        Samples the input point cloud by a ratio percentage to reduce
+        cardinality and uses an mlp to augment features dimensionnality
+    '''
+
+    def __init__(self, in_channels, out_channels, ratio=0.25, k=16):
+        super(TransitionDown, self).__init__()
+        self.k = k
+        self.ratio = ratio
+        self.mlp = Seq(
+            Lin(in_channels, out_channels),
+            BN(out_channels),
+            ReLU()
+        )
+
+    def forward(self, x, pos, batch):
+        # FPS sampling
+        id_clusters = fps(pos, ratio=self.ratio, batch=batch)
+
+        # compute for each cluster the k nearest points
+        sub_batch = batch[id_clusters] if batch is not None else None
+
+        # beware of self loop
+        id_k_neighbor = knn(pos, pos[id_clusters], k=self.k,
+                            batch_x=batch, batch_y=sub_batch)
+
+        # transformation of features through a simple MLP
+        x = self.mlp(x)
+
+        # Max pool onto each cluster the features from knn in points
+        x_out, _ = scatter_max(x[id_k_neighbor[1]],
+                               id_k_neighbor[0],
+                               dim_size=id_clusters.size(0),
+                               dim=0)
+
+        # keep only the clusters and their max-pooled features
+        new_pos, new_x = pos[id_clusters], x_out
+        return new_x, new_pos, sub_batch
+
+
+class Net(torch.nn.Module):
+    def __init__(self, NUM_CLASSES, NUM_FEATURES, dim_model, k=16):
+        super(Net, self).__init__()
+        self.k = k
+
+        # dummy feature is created if there is none given
+        NUM_FEATURES = max(NUM_FEATURES, 1)
+
+        # first block
+        self.mlp_input = Seq(
+            Lin(NUM_FEATURES, dim_model[0]),
+            BN(dim_model[0]),
+            ReLU()
+        )
+
+        self.transformer_input = TransformerBlock(in_channels=dim_model[0],
+                                                  out_channels=dim_model[0])
+        # backbone layers
+        self.transformers_down = torch.nn.ModuleList()
+        self.transition_down = torch.nn.ModuleList()
+
+        for i in range(len(dim_model) - 1):
+            # Add Transition Down block followed by a Transformer block
+            self.transition_down.append(
+                TransitionDown(in_channels=dim_model[i],
+                               out_channels=dim_model[i + 1],
+                               k=self.k)
+            )
+
+            self.transformers_down.append(
+                TransformerBlock(in_channels=dim_model[i + 1],
+                                 out_channels=dim_model[i + 1])
+            )
+
+        # class score computation
+        self.lin = Seq(Lin(dim_model[-1], 64),
+                       ReLU(),
+                       Lin(64, 64),
+                       ReLU(),
+                       Lin(64, NUM_CLASSES))
+
+    def forward(self, data):
```

Comment:
```suggestion
    def forward(self, x, pos, batch=None):
```
Let's make required arguments explicit.

Reply:
I feel like putting data would allow models to be more independant, like bricks that we can move from one code to another. Is there a good practice consensus on this, wether in PyG or in Pytorch ? I took as reference the example on PointNet in PyG : https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pointnet2_segmentation.py

Reply:
If you feel that you would leave the input args as `data` I'm fine with that as well. I know that we used `data` as input argument in previous examples extensively, but I consider it bad-practice for now, since it is not really clear which attributes of the `data` needs to be set.

An alternative way is to define `forward` as:
```python
def forward(self, x, pos, batch=None, *args, **kwargs):
    pass
```
and call it via
```python
model(**data.to_dict())
```